vllm/scripts.py

```diff
@@ -0,0 +1,193 @@
+# The CLI entrypoint to vLLM.
+import argparse
+import requests
+from typing import Dict, Any
+
+from vllm.entrypoints.openai.api_server import run_server
+from vllm.entrypoints.openai.cli_args import make_arg_parser
+from vllm.entrypoints.openai.protocol import (CompletionRequest, ChatCompletionRequest)
+
+COMPLETE_ROUTE = "/v1/completions"
+CHAT_COMPLETE_ROUTE = "/v1/chat/completions"
+LIST_MODELS_ROUTE = "/v1/models"
+
+def main():
+    parser = argparse.ArgumentParser(description="vLLM CLI")
+    subparsers = parser.add_subparsers(required=True)
+
+    serve_parser = subparsers.add_parser(
+        "serve",
+        help="Start the vLLM OpenAI Compatible API server",
+        usage="vllm serve <model_tag> [options]")
+    # Override the `--model` optional argument, make it positional.
+    serve_parser.add_argument(
+        "model-tag",
+        type=str,
+        help="The model tag to serve")
+    serve_parser = make_arg_parser(serve_parser)
+    serve_parser.set_defaults(func=run_server)
+    
+    complete_parser = subparsers.add_parser(
+        "complete",
+        help="Generate text completions based on the given prompt via the running API server",
+        usage="vllm complete [options]")
+    complete_parser.add_argument("--url",
+                                 type=str,
+                                 default=None,
+                                 help="url of the running OpenAI-Compatible RESTful API server")
+    complete_parser.add_argument("--complete-model-name",
+                                type=str, default=None,
+                                help="the model name used in prompt completion")
+    complete_parser.set_defaults(func=interactive_cli)
+
+    chat_parser = subparsers.add_parser(
+        "chat",
+        help="Generate chat completions via the running API server",
+        usage="vllm chat [options]"
+    )
+    chat_parser.add_argument("--url",
+                                 type=str,
+                                 default=None,
+                                 help="url of the running OpenAI-Compatible RESTful API server")
+    chat_parser.add_argument("--chat-model-name",
+                                type=str, default=None,
+                                help="the model name used in chat completions")
+    chat_parser.add_argument("--generation-prompt",
+                             type=str,
+                             help="the generation prompt to be added to the chat template")
```

Comment:
```suggestion
```

Reply:
However, let's add an option for `--system-prompt` so for method that support system prompt user can put in one.